src/key_value_store/core/KeyValueStore.cpp

```diff
@@ -0,0 +1,66 @@
+#include "KeyValueStore.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+// Alias for the unorderedList
+typedef std::unordered_map<std::string, 
+  std::vector<std::string>> hashMap; 
+
+bool KeyValueStore::Put(std::string key, std::string value) {
+  bool result; 
+  lock_.lock();
+  // If this key already exists in the store
+  if (kv_store_.find(key) != kv_store_.end()) {
+    // If user puts the same function
+    if (CheckIfExists(kv_store_[key], value)) {
+      result = false;
+    } else {
+      int size = kv_store_[key].size();
+      kv_store_[key].push_back(value);
+      // If the value was not inserted
```

Comment:
Generally you don't need to check whether push_back works -- if it fails, it's because you ran out of memory so you probably going to crash.

Reply:
Okay, removed.